docs/samples/opera/app/types.d.ts

```diff
@@ -77,6 +77,79 @@ export interface CatalogState {
     status: CatalogStatus;
 }
 
+export interface BindingSpec {
+    role: string;
+    provider: string;
+    config: Record<string, string>;
+}
+
+export interface TopologySpec {
+    device: string;
+    selector: Record<string, string>;
+    bindings: BindingSpec[];
+}
+
+export interface TargetSpec {
+    displayName: string;
+    scope: string;
+    metadata: Record<string, string>;
+    properties: Record<string, string>;
+    components: ComponentSpec[];
+    constraints: string;
+    topologies: TopologySpec[];
+    forceRedeploy: boolean;
+    generation: string;
+    version: string; 
+}
+
+export interface ComponentError {
+    code: string;
+    message: string;
+    target: string;
+}
+
+export interface TargetError {
+    code: string;
+    message: string;
+    target: string;
+    details: ComponentError[];
+}
+
+export interface ErrorType {
+    code: string;
+    message: string;
+    target: string;
+    details: TargetError[];
+}
+
+export interface ProvisioningStatus{
+    operationId: string;
+    status: string;
+    failureCause: string;
+    logErrors: boolean;
+    error: ErrorType;
+    output: Record<string, string>;
+}
+
+export interface DeployableStatus {
+    properties: Record<string, string>;
+    ProvisioningStatus: ProvisioningStatus;
+    lastModified: Date;
+}
+
+export interface ObjectMeta {
+    namespace: string;
+    name: string;
+    labels: Record<string, string>;
+    annotations: Record<string, string>;
+}
+
+export interface TargetState {
```

Comment:
I saw in this file, there's still some object like SolutionState which doesn't align new object model. Do we need to fix them?